django/views/generic/list.py

```diff
@@ -63,8 +63,13 @@ def paginate_queryset(self, queryset, page_size):
         Paginate the queryset, if needed.
         """
         paginator = self.get_paginator(
-            queryset, page_size, orphans=self.get_paginate_orphans(),
-            allow_empty_first_page=self.get_allow_empty())
+            queryset,
+            page_size,
+            orphans=self.get_paginate_orphans(),
+            allow_empty_first_page=self.get_allow_empty(),
+            request=self.request,
+            page_kwarg=self.page_kwarg
```

Comment:
please include trailing comma so if more items are added later we don't need to modify this line again